keras_preprocessing/image/affine_transformations.py

```diff
@@ -325,6 +345,12 @@ def apply_affine_transform(x, theta=0, tx=0, ty=0, shear=0, zx=1, zy=1,
         transform_matrix = transform_matrix_offset_center(
             transform_matrix, h, w)
         x = np.rollaxis(x, channel_axis, 0)
+
+        # Matrix construction assumes that coordinates are x, y (in that order).
+        # However, regular numpy arrays use y,x (aka i,j) indexing.
+        # Swap the first two columns of the matrix to simulate the x, y indexing.
+        if col_axis > row_axis:
```

Comment:
So this happens if the user has a shape channel-col-row (CWH) right?
If this is the case, I don't think we should support it.

Reply:
On the contrary, it happens in the most usual cases CHW and HWC.

Currently, ```apply_affine_transform(x, tx=1)``` will move image in the **vertical** direction rather than horizontal. Similarly, ```apply_affine_transform(x, ty=1)`` will move image in the **horizontal** direction rather than vertical.

I added unit tests that demonstrate this behavior.